src/main.rs

```diff
@@ -1,126 +1,267 @@
-use std::fs::File;
-use std::io::Error as IOError;
+mod cel;
+mod light;
+mod outline;
+mod color;
+mod scale;
+mod geometry;
+mod material;
 
-use fbx_direct::reader::{EventReader, FbxEvent::*, Error as FBXError};
-use nalgebra::Point2D; //TODO: Add the nalgebra crate and lookup the closest struct to Point2D
+use std::f32::consts::PI;
+use std::time::Duration;
+use std::path::Path;
+use std::thread;
+use std::rc::Rc;
 
-// TODO: Pick a better name than AppError
-#[derive(Debug)]
-enum AppError {
-    IOError(IOError),
-    FBXError(FBXError),
+use euc::{Pipeline, rasterizer, buffer::Buffer2d, Target};
+use minifb::{self, Key, KeyRepeat};
+use tobj;
+use vek::{Mat4, Vec2, Vec3, Vec4, Rgba};
+use image::ImageBuffer;
+
+use crate::cel::CelShader;
+use crate::outline::OutlineShader;
+use crate::light::DiffuseLight;
+use crate::scale::scale_buffer;
+use crate::geometry::Mesh;
+use crate::material::Material;
+use crate::color::{rgba_to_bgra_u32, bgra_u32_to_rgba, vek_rgba_to_image_rgba};
+
+fn main() {
+    let image_width = 64;
```

Comment:
I know this is a prototype but we should probably parameterize this so we can see what the output's like for other resolutions (ex: 32 x 32)

Reply:
Agreed!